packages/react-instantsearch-dom/src/components/__tests__/PanelCallbackHandler.js

```diff
@@ -22,101 +19,87 @@ describe('PanelCallbackHandler', () => {
 
   describe('willMount', () => {
     it('expect to call setCanRefine when the context is given', () => {
-      const props = {
-        canRefine: true,
-      };
-
-      const context = {
-        setCanRefine: jest.fn(),
-      };
-
-      shallow(
-        <PanelCallbackHandler {...props}>
-          <div>Hello content</div>
-        </PanelCallbackHandler>,
-        { context }
+      const setCanRefine = jest.fn();
+
+      mount(
+        <PanelProvider value={setCanRefine}>
+          <PanelCallbackHandler canRefine={true}>
+            <div>Hello content</div>
+          </PanelCallbackHandler>
+        </PanelProvider>
       );
 
-      expect(context.setCanRefine).toHaveBeenCalledTimes(1);
-      expect(context.setCanRefine).toHaveBeenCalledWith(true);
+      expect(setCanRefine).toHaveBeenCalledTimes(1);
+      expect(setCanRefine).toHaveBeenCalledWith(true);
     });
 
     it('expect to not throw when the context is not given', () => {
-      const props = {
-        canRefine: true,
-      };
-
-      const trigger = () =>
```

Comment:
I found it easier to read with the `trigger` function but you can stick with the inline form.